fhir-model/src/main/java/com/ibm/fhir/model/util/ValidationSupport.java

```diff
@@ -355,4 +357,37 @@ public static void prohibited(Element element, String elementName) {
             throw new IllegalStateException(String.format("Element: '%s' is prohibited.", elementName));
         }
     }
+    
+    /**
+     * @throws IllegalStateExeption if the codeableConcept has coding elements that do not include codes from the required binding
+     */
+    public static void checkCodeableConcept(CodeableConcept codeableConcept, String elementName, String system, String... codes) {
+        if (codeableConcept != null && !codeableConcept.getCoding().isEmpty() && hasCodingWithSystemAndCodeValues(codeableConcept)) {
+            List<String> codeList = Arrays.asList(codes);
+            for (Coding coding : codeableConcept.getCoding()) {
+                if (hasSystemAndCodeValues(coding) && 
+                        system.equals(coding.getSystem().getValue()) && 
+                        codeList.contains(coding.getCode().getValue())) {
+                    return;
+                }
+            }
+            throw new IllegalStateException(String.format("Element: '%s' must contain a valid code from system: '%s'", elementName, system));
```

Comment:
isn't it more about containing a valid value from a given valueset?
anyway, i guess all the valuesets in question here are "identity" valuesets for a given codesystem, so probably its fine.

Reply:
Instances are concerned with systems but from an error message perspective, I could see how that could be improved.

Reply:
Reversed course and put value set in error message instead of code system.